### PR TITLE
[SYCL] Mark arguments as potentially unused

### PR DESCRIPTION
**Pull request by sommerlukas**
Fix compilation failure in post-commit CI due to some arguments currently being unused.

### CODE REVIEW

**Review by mfrancepillois — APPROVED** [review]

File: sycl/source/detail/jit_compiler.cpp [hunk 1]
```diff
@@ -153,7 +153,8 @@ struct PromotionInformation {
 
 using PromotionMap = std::unordered_map<SYCLMemObjI *, PromotionInformation>;
 
-template <typename Obj> Promotion getPromotionTarget(const Obj &obj) {
+template <typename Obj>
+Promotion getPromotionTarget([[maybe_unused]] const Obj &obj) {
```
Comment (FMarno) [inline sycl/source/detail/jit_compiler.cpp:157]: Can you not delete the parameter name instead? since it is actually unused.
same for the other change

**Review by victor-eds — APPROVED** [review]
LGTM.

File: sycl/source/detail/jit_compiler.cpp (hunk 1 above)
Comment (victor-eds) [reply sycl/source/detail/jit_compiler.cpp:157]: I agree with this, but didn't wanna be too picky

Comment (sommerlukas) [reply sycl/source/detail/jit_compiler.cpp:157]: We could, but we expect to need this parameter in another PR soonish, so this change seems less invasive and allows us to fix post-commit CI quicker.

**Review by FMarno — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]

**Comment by steffenlarsen** [comment]
Precommit builds passed and testing should be unaffected. Merging while tests are running.

**Pull request merged**